$OpenBSD: patch-src_filter_plugins_ReplayGainFilterPlugin_cxx,v 1.1 2016/07/04 08:50:28 dcoppa Exp $

commit 072e39c9cf449ccb8d7be301768a769f479c2875
Author: Max Kellermann <max@duempel.org>
Date:   Fri Jul 1 21:17:52 2016 +0200

filter/ReplayGain: skip PcmVolume if a mixer is set

Previously, volume was applied twice: once by PcmVolume, and again by
the hardware mixer.

--- src/filter/plugins/ReplayGainFilterPlugin.cxx.orig	Fri Dec 26 14:15:21 2014
+++ src/filter/plugins/ReplayGainFilterPlugin.cxx	Mon Jul  4 10:34:23 2016
@@ -134,8 +134,6 @@ ReplayGainFilter::Update()
 		volume = pcm_float_to_volume(scale);
 	}
 
-	pv.SetVolume(volume);
-
 	if (mixer != nullptr) {
 		/* update the hardware mixer volume */
 
@@ -146,7 +144,8 @@ ReplayGainFilter::Update()
 		Error error;
 		if (!mixer_set_volume(mixer, _volume, error))
 			LogError(error, "Failed to update hardware mixer");
-	}
+	} else
+		pv.SetVolume(volume);
 }
 
 static Filter *
@@ -174,7 +173,9 @@ ReplayGainFilter::Close()
 ConstBuffer<void>
 ReplayGainFilter::FilterPCM(ConstBuffer<void> src, gcc_unused Error &error)
 {
-	return pv.Apply(src);
+	return mixer != nullptr
+		? src
+		: pv.Apply(src);
 }
 
 const struct filter_plugin replay_gain_filter_plugin = {
